iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/plan/node/source/TimeseriesRegionScanNode.java

```diff
@@ -31,25 +33,37 @@
 import org.apache.iotdb.db.queryengine.plan.planner.plan.node.PlanVisitor;
 
 import com.google.common.collect.ImmutableList;
+import org.apache.tsfile.file.metadata.PlainDeviceID;
 import org.apache.tsfile.utils.ReadWriteIOUtils;
 
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class TimeseriesRegionScanNode extends RegionScanNode {
-  private Map<PartialPath, TimeseriesSchemaInfo> timeseriesToSchemaInfo;
+  // DevicePath -> (MeasurementPath -> TimeseriesSchemaInfo), it will be serialized to
+  // timeseriesToSchemaInfo in the end
+  Map<PartialPath, Map<PartialPath, List<TimeseriesSchemaInfo>>> deviceToTimeseriesSchemaInfo;
```

Comment:
add private

Reply:
what if we don't call serialize and deserialize method in which case that DataRegion is in current node and we directly pass this object.